config.m4

```diff
@@ -173,18 +184,18 @@ if test "$PHP_MONGODB" != "no"; then
     AC_MSG_CHECKING(for libbson)
     if test -x "$PKG_CONFIG" && $PKG_CONFIG --exists libbson-1.0; then
       if $PKG_CONFIG libbson-1.0 --atleast-version 1.9.0; then
-        LIBBSON_INC=`$PKG_CONFIG libbson-1.0 --cflags`
-        LIBBSON_LIB=`$PKG_CONFIG libbson-1.0 --libs`
-        LIBBSON_VER=`$PKG_CONFIG libbson-1.0 --modversion`
-        AC_MSG_RESULT(version $LIBBSON_VER found)
+        PHP_MONGODB_BSON_CFLAGS=`$PKG_CONFIG libbson-1.0 --cflags`
+        PHP_MONGODB_BSON_LIBS=`$PKG_CONFIG libbson-1.0 --libs`
+        PHP_MONGODB_BSON_VERSION=`$PKG_CONFIG libbson-1.0 --modversion`
+        AC_MSG_RESULT(version $PHP_MONGODB_BSON_VERSION found)
       else
         AC_MSG_ERROR(system libbson must be upgraded to version >= 1.9.0)
       fi
     else
       AC_MSG_ERROR(pkgconfig and libbson must be installed)
     fi
-    PHP_EVAL_INCLINE($LIBBSON_INC)
-    PHP_EVAL_LIBLINE($LIBBSON_LIB, MONGODB_SHARED_LIBADD)
+    PHP_MONGODB_CFLAGS="$PHP_MONGODB_CFLAGS $PHP_MONGODB_BSON_CFLAGS"
```

Comment:
(Not important, if this is broken, people will tell us)

Reply:
`PHP_EVAL_INCLINE` calls `PHP_ADD_INCLUDE` for each `-I` path, which calls both `PHP_EXPAND_PATH` (expanding to an absolute path) and `PHP_RUN_ONCE` to ensure each path is only added once.

Nearly all of these instances obtain CFLAGS from `pkg-config` being called for a unique library, so I'm comfortable trusting that their CFLAGS can be added as-is. The only overlap IMO is libmongoc and libbson. Since libmongoc's `libmongoc-1.0.pc` file depends on `libbson-1.0`, it ends up including libbson's `pkg-config` output in its own. To address that, I suppose we could skip merging in libbson's own `pkg-config` output and just rely on libmongoc. I'd propose making this change today.

One real exception is `AX_PTHREAD`, which is likely going to return `-pthread` (ignored by `PHP_EVAL_INCLINE` entirely) instead of an `-I` path. For that, I think we certainly need to just append to `PHP_MONGODB_CFLAGS`.

One other thought. There are more `pkg-config` options than I realized. Per [`pkg-config(1)`](https://linux.die.net/man/1/pkg-config), we can request `-I` CFLAGS and "other" CFLAGS independently, which would let us rely on `PHP_EVAL_INCLINE` and defer to `PHP_MONGODB_CFLAGS` when necessary. Similarly, we can requires `-L` and -l` libs independently from "other" linker arguments (which might require manual appends instead of `PHP_EVAL_LIBLINE`). While this is trivial to add for libbson and libmongoc, since we call `pkg-config` directly, we also know by looking at their package files that it would have no functional impact. It _could_ be helpful for other libraries (e.g. `CheckSSL.m4`), but those are all using the `pkg.m4` macro at the moment and would require some significant refactoring. I don't think there's a pressing need for this, but it does seem like it'd be the "correct" approach. If it sounds worthwhile, I can make a separate ticket to track the change and we can triage it for some future release.

Reply:
I wouldn't worry about it then. I'd defer making a ticket until somebody notices it.